### PR TITLE
Support scaling input features to attacks 

### LINKED ISSUE #2152: Support feature scaling in membership black-box attack
Membership inference black-box attack trains an attack model on both true labels and predictions or losses. At least for the rf and gb attack model types, these are simply concatenated and fed into the model. Scaling these features can help the attack model to be more accurate.

Requested solution:
Apply scaling (using different options of standard sklearn scalers) on the input features to the attack before training and inferring.

### PR DESCRIPTION
**Pull request by abigailgold**
# Description

Both black-box membership inference and black-box attribute inference can scale input features to the attack using multiple options of sciki-learn scalers (default is StandardScaler).

Fixes #2152

## Type of change

Please check all relevant options.

- [X] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

New tests that try out all options for scaling.

**Test Configuration**:
- OS: MacOS 14.2.1
- Python version: 3.9
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] My changes have been tested using both CPU and GPU devices

### CODE REVIEW

**Review by beat-buesser — APPROVED** [review]
Hi @abigailgold Thank you very much! It looks good to me.
